core/src/iam/signin.rs

```diff
@@ -141,10 +141,11 @@ pub async fn db_access(
 									match val.record() {
 										// There is a record returned
 										Some(rid) => {
+											let mut rid = rid;
 											// Create the authentication key
 											let key = config(iss.alg, iss.key)?;
 											// Create the authentication claim
-											let val = Claims {
+											let mut claims = Claims {
```

Comment:
I don't think we should touch the token as provided by the user. Do you see any specific benefit from updating the claims here? Since we are only updating the `id` claim, I think access to `$auth.id` is sufficient.

Reply:
I addressed this in [a54b5bd](https://github.com/surrealdb/surrealdb/pull/4161/commits/a54b5bd5c4a8b4817a71b8f90a2cf8ba87ea0949). Let me know what you think!